### ISSUE TITLE
output file duration is 0

### ISSUE THREAD

**Issue by wfalkwallace** [dbridges/cocoa-tlassemble#3]
I'm running tlassemble (from homebrew) on OSX mavericks with all the default options and I keep getting a broken output file with zero-duration. Any ideas? homebrew should handle dependencies, right? could I be missing necessary encoders?

**Comment by wfalkwallace** [comment]
UPDATE: it only happens for some sets of images - could it have to do with filesizes? The only difference in the image sets that I can see is the filesize/image-size
- one folder (that works) has images that are 507kb (2560x1440)
- the other (that doesn't work) has images that are 2.9mb (4128x2322)

**Comment by dbridges** [comment]
Yes, you are absolutely correct about the image size. This seems to be a limitation by the Quicktime framework, so there is not much to be done at this point.

**Comment by wfalkwallace** [comment]
any idea what that limit is? any way around this, or are you using Quicktime for the conversion/construction?

**Comment by wadetregaskis** [comment]
> any idea what that limit is? any way around this, or are you using Quicktime for the conversion/construction?

Unfortunately it seems to be a fundamental MacOS X limit.  I forget now exactly the limit factors, but with a 3:2 aspect ratio at least, the maximum height you can [reliably] use with the Quicktime framework is 2512.  If you use AVFoundation it's slightly less, at 2496.  IIRC it prefers multiples of four or eight or somesuch, so you can sometimes inch slightly above these (e.g. 2504, 2516, etc) depending on some indeterminate other factors.  You're using a 16:9 ratio, so it's not surprising a height of 2322 fails - presumably the underlying limitation is on some frame buffer size, so it's a function of the total area, not specifically the height (and 4128x2322 is slightly larger in area than 3768x2512).

Assuming it's purely area, I estimate your limit at 16:9 is about 4096x2304.

ffmpeg and other open-source encoders do not have these limits (and are in many other ways superior to Apple's codecs), but I've not yet tried integrating them [into my fork of tlassemble].  It's on my [long] TODO list.

**Comment by nathanbeach** [comment]
I noticed this happening only with tlassemble, but not with the Time Lapse Assembler UI in OSX.  That is, I can successfully assemble a given set of images with the UI, but the output file is zero if I try to do the same thing at the command line with tlassemble.  I'm trying to assemble upwards of 4000 Foscam security camera images which are pretty poorly put together jpegs (they don't even have exif data and some get warnings when I'm doing some pre-processing with ImageMagick).  But I've also tried a smaller subset of the images and it doesn't work either.  Will update this thread if I figure anything else out.  Thanks!
